src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -146,28 +156,55 @@ protected async Task WaitMessageIntervalDuringRecoveryAsync(CancellationToken ca
             Logger.LogDebug("Circuit breaker caused message pump '{JobId}' to wait message interval during recovery of '{Interval}' during the '{State}' state", JobId, CircuitState.Options.MessageIntervalDuringRecovery.ToString("g"), CircuitState);
             await Task.Delay(CircuitState.Options.MessageIntervalDuringRecovery, cancellationToken);
 
+            MessagePumpCircuitState oldState = CircuitState;
+            bool shouldTriggerTransitionEvent = !oldState.IsHalfOpen;
+
             CircuitState = CircuitState.TransitionTo(CircuitBreakerState.HalfOpen);
+
+            if (shouldTriggerTransitionEvent)
```

Comment:
Don't notify when we go from half-open to half-open.

Reply:
Actually, is it even necessary to call `TransitionTo` if we remain in half-open state ?